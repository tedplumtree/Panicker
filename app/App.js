import React from 'react';
import { ThemeProvider } from 'react-native-material-ui';
import { Provider as StoreProvider } from 'react-redux';
import { createStore } from 'redux';
import { MaterialDesignTheme } from './config/styles';
import SettingsAndActivateScreen from './screens/SettingsAndActivateScreen';
import combinedReducers from './reducers/index';

const store = createStore(combinedReducers);

const App = () => (
  <StoreProvider store={store}>
    <ThemeProvider uiTheme={MaterialDesignTheme}>
      <SettingsAndActivateScreen />
    </ThemeProvider>
  </StoreProvider>
);

export default App;
