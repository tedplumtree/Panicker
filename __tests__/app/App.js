import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import App from '../../app/App';

jest.mock('react-native-material-ui', () => ({
  // Force ThemeProvider to render a component by mocking it, so we can test it
  ThemeProvider: 'MockThemeProvider',
  COLOR: {
    amber500: 'dummy color',
  },
}));

jest.mock('react-redux', () => ({
  // Force Provider to render a component by mocking it, so we can test it
  Provider: 'MockStoreProvider',
}));

jest.mock('../../app/screens/SettingsAndActivateScreen', () => (
  'MockSettingsAndActivateScreen'
));

it('sets up the app and renders correctly', () => {
  const tree = renderer.create(
    <App />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
