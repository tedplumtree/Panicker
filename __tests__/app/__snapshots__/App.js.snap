// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets up the app and renders correctly 1`] = `
<MockStoreProvider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <MockThemeProvider
    uiTheme={
      Object {
        "palette": Object {
          "primaryColor": "dummy color",
        },
      }
    }
  >
    <MockSettingsAndActivateScreen />
  </MockThemeProvider>
</MockStoreProvider>
`;
